tower-scripts/bin/cicd-control.sh

```diff
@@ -282,28 +338,52 @@ elif [ "${OPERATION}" == "upgrade" ]; then
     export AWS_DEFAULT_PROFILE=$AWS_ACCOUNT_NAME
     export SKIP_GIT_VALIDATION=TRUE
     /usr/local/bin/autokeys_loader ./aws_online_cluster_upgrade.sh ./ops-to-productization-inventory.py ${CLUSTERNAME}
-  popd
+}
 
 ################################################
-# PERFORMANCE TEST1
+# OPERATION: GENERAL OPERATIONS CLUSTER
 ################################################
-elif [ "${OPERATION}" == "perf1" ]; then
+function cluster_operation() {
 
-    if [[ "${CLUSTERNAME}" == "test-key" ]]; then
-        echo "Mock run for: ${CLUSTERNAME}"
-        exit 0
-    fi
+  if [ -z "${1+x}" ]; then
+    echo "No UPGRADE OPTION was specified.  Exiting..."
+    exit 1
+  fi
 
-    if [[ "${CLUSTERNAME}" != "free-int" && "${CLUSTERNAME}" != "dev-preview-int" ]]; then
-        echo "Cannot run performance test on cluster: ${CLUSTERNAME}"
-        exit 1
-    fi
+  UPGRADE_OPERATION=$1
+
+  echo "Doing upgrade operation: ${UPGRADE_OPERATION}"
+
+  # Do long running operations
+  is_running &
+  ./disable-docker-timer-hack.sh "${CLUSTERNAME}" > /dev/null &
+
+  # setup cluster vars
+  setup_cluster_vars
 
-    is_running & 
-    echo "Running performance test 1"
-    MASTER="$(get_master_name)"
+  # Get the latest openshift-ansible rpms
+  get_latest_openshift_ansible ${oo_environment}
+
+  # Run the upgrade, including post_byo steps and config loop
+  pushd ~/aos-cd/git/openshift-ansible-ops/playbooks/release/bin
+    /usr/local/bin/autokeys_loader ./cicd_operations.sh -i ./cicd-to-productization-inventory.py -c ${CLUSTERNAME} -o ${UPGRADE_OPERATION}
```

Comment:
inventory probably needs to be generated by the operation in cicd_operations (e.g. inventory for master-only upgrade differs from node-only upgrade).